### DISCUSSION TITLE
Enable building openssl from source automatically or using command-line flag

### DISCUSSION THREAD

**Discussion by sureshvv** [pyenv/pyenv#2917] [Ideas] [answered]
Have had to do this repeatedly on different systems.

**Comment by native-api** [comment, marked as answer]
This is not currently supported. If you need to build additional packages from source or otherwise create a custom setup, you are supposed to do that as a separate step before calling `pyenv install` and provide additional compiler flags to Pyenv if necessary.

The functionality to build OpenSSL and Readline from source was added to only cover the specific case of MacOS without 3rd-party versions of those libraries -- because in that case, those dependencies are present (thus system requirements are technically met) but cannot be used due to Apple's customizations.
